pandas/tests/scalar/interval/test_interval.py

```diff
@@ -222,3 +265,12 @@ def test_constructor_errors_tz(self, tz_left, tz_right):
         error = TypeError if com._any_none(tz_left, tz_right) else ValueError
         with pytest.raises(error):
             Interval(left, right)
+
+    def test_overlaps(self, interval_overlaps_params):
```

Comment:
This should be moved to `tests/scalar/interval/test_ops.py`, as that's where the existing tests for `overlaps` are.

Some of the test cases you created are already covered there, and for other cases you can use some of the existing machinery there to make things more concise.  Something like:
```python
def test_overlaps_interval_empty(self, start_shift, closed, other_closed):
    # GH26893
    start, shift = start_shift
    interval1 = Interval(start, start + 2 * shift, closed)
    interval2 = Interval(start + shift, start + shift, other_closed)

    # overlap only if interval2 is a point (i.e. not empty)
    result = interval1.overlaps(interval2)
    expected = other_closed == 'both'
    assert result is expected

def test_overlaps_empty_empty(self, start_shift, closed, other_closed):
    # GH26893
    start = start_shift[0]
    interval1 = Interval(start, start, closed)
    interval2 = Interval(start, start, other_closed)

    # overlap only if both are points (i.e. not empty)
    result = interval1.overlaps(interval2)
    expected = closed == other_closed == 'both'
    assert result is expected
```

Reply:
Moved to `test_ops`, but I kept the separate fixture which I find easier to read and caught a lot of bugs when I tested alternative fixes. I also don't want to cut and paste the cases, but share them between array/scalar, so that adding a case applies to both. I only discovered  `IntervalArray` uses its own `overlaps` code by mistake. If the tests were shared,
adding a case would have tripped the bug in both.